lib/tasks/add_missing_asset_audit_records.rake

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+namespace :asset_audit do
+  desc 'Add missing asset audit records'
+  task :add_missing_records, [:file_path] => :environment do |_, args|
+    file_path = args[:file_path] == 'nil' ? nil : args[:file_path]
+    if file_path.nil? || !File.exist?(file_path)
+      puts 'Please provide a valid file path'
+      next
+    end
+
+    puts 'Adding missing asset audit records...'
+
+    ActiveRecord::Base.transaction do
+      begin
+        csv_data = CSV.read(file_path, headers: true)
+      rescue StandardError => e
+        puts "Failed to read CSV file: #{e.message}"
+        next
+      end
+      csv_data.each do |row|
+        asset = Labware.find_by_barcode(row['barcode'].strip)
+        next if asset.nil?
+
+        key =
+          case row['message']
+          when 'Destroying location'
+            'destroy_location'
+          when 'Destroying labware'
+            'destroy_labware'
+          end
+        next if key.nil?
+
+        begin
+          AssetAudit.create!(
+            message: "Process '#{row['message']}' performed on instrument Destroying instrument",
+            created_by: row['created_by'].strip,
+            created_at: row['created_at'].strip,
+            asset_id: asset.id,
+            key: key
+          )
+          puts "Record for asset_id #{asset.id} successfully inserted."
+        rescue ActiveRecord::ActiveRecordError, StandardError => e
+          puts "Error inserting record for asset_id #{asset.id}: #{e.message}"
```

Comment:
Because you are catching and handling the exception here, the transaction block becomes useless. If we do it like this, we will need to record a log and deal with each individual case separately to fix them. I suppose the intention was rolling back the whole transaction if there is even a single error. Either let the exception bubble up or explicitly rollback the active transaction and break the loop.